packages/connect-ui/src/views/Passphrase.tsx

```diff
@@ -31,15 +32,32 @@ const Divider = styled.div`
     background: ${props => props.theme.STROKE_GREY};
 `;
 
-export type PassphraseProps = Extract<UiEvent, { type: 'ui-request_passphrase' }> & {
-    onPassphraseSubmit: (value: string, enterOnDevice?: boolean) => void;
-};
+export type PassphraseEventProps = Extract<UiEvent, { type: 'ui-request_passphrase' }>;
+
+type PassphraseProps = PassphraseEventProps & { postMessage: any };
 
 export const Passphrase = (props: PassphraseProps) => {
-    const { onPassphraseSubmit } = props;
     const { device } = props.payload;
     const { features } = device;
 
+    const onPassphraseSubmit = (value: string, passphraseOnDevice?: boolean) => {
+        props.postMessage(
+            createUiResponse(UI.RECEIVE_PASSPHRASE, {
+                value,
+                passphraseOnDevice,
+                // todo: what is this param?
+                save: true,
+            }),
+        );
+
+        analytics.report({
+            type: EventType.WalletType,
+            payload: {
+                type: value ? 'hidden' : 'standard',
+            },
+        });
+    };
```

Comment:
discussion: this postMessage can be caught in iframe exactly here
https://github.com/trezor/trezor-suite/blob/feat/connect-popup-analytics/packages/connect-iframe/src/index.ts#L45
wouldn't it make sense to have analytics centralized in this place? Lets talk about pros and cons:

### Pros:
- centralized, easy to follow what is being tracked.

### Cons: 
- messages are atomic, can't log additional context without introducing some kind of new local state (we don't want that). but do we need  that? for example, in this case, we don't